standalone-metastore/metastore-common/src/main/thrift/hive_metastore.thrift

```diff
@@ -2454,10 +2467,14 @@ PartitionsResponse get_partitions_req(1:PartitionsRequest req)
   // partition keys in new_part should be the same as those in old partition.
   void rename_partition(1:string db_name, 2:string tbl_name, 3:list<string> part_vals, 4:Partition new_part)
                        throws (1:InvalidOperationException o1, 2:MetaException o2)
-  
+
   RenamePartitionResponse rename_partition_req(1:RenamePartitionRequest req)
                        throws (1:InvalidOperationException o1, 2:MetaException o2)
 
+  // Returns a file list using FlatBuffers as serialization
+  GetFileListResponse get_file_list(1:GetFileListRequest req)
+    throws(1:MetaException o1)
```

Comment:
Seems like if the partition doesn't exist you might need to throw a NoSuchObjectFoundException as well.

Reply:
You're right, I forgot to add it.